src/main/java/org/openstreetmap/atlas/checks/validation/areas/PedestrianAreaOverlappingEdgeCheck.java

```diff
@@ -0,0 +1,230 @@
+     */
+    private boolean isNotSnappedToSegments(final List<Segment> segments, final Location locationOne,
+            final Location locationTwo)
+    {
+        return segments.stream().noneMatch(segment -> (locationOne.equals(segment.start())
+                || locationOne.equals(segment.end()))
+                && (locationTwo.equals(segment.start()) || locationTwo.equals(segment.end())));
+    }
+
+    /**
+     * Checks if the given edge and area have the same elevation or not.
+     *
+     * @param edge
+     *            any given edge
+     * @param area
+     *            any given area
+     * @return true if the edge and area have the same Layer and Location tag values
+     */
+    private boolean isOfSameElevation(final Edge edge, final Area area)
+    {
+        return area.getTag(LayerTag.KEY).orElse("").equals(edge.getTag(LayerTag.KEY).orElse(""))
+                && area.getTag(LocationTag.KEY).orElse("")
+                        .equals(edge.getTag(LocationTag.KEY).orElse(""))
+                && area.getTag(LevelTag.KEY).orElse("")
+                        .equals(edge.getTag(LevelTag.KEY).orElse(""));
+    }
+
+    /**
+     * Checks if given {@link AtlasObject} is a pedestrian area or not. A pedestrian area is an
+     * {@link AtlasObject} with "area=yes" and "highway=pedestrian" tags.
+     *
+     * @param object
+     *            any {@link AtlasObject}
+     * @return true if the object is a pedestrian area
+     */
+    private boolean isPedestrianArea(final AtlasObject object)
+    {
+        return Validators.isOfType(object, AreaTag.class, AreaTag.YES)
+                && Validators.isOfType(object, HighwayTag.class, HighwayTag.PEDESTRIAN);
+    }
+
+    /**
+     * Checks if an intersecting edge has a different osm id than the area, does not have pedestrian
+     * highway tags, is car navigable and is of the same level or layer as the area.
+     *
+     * @param edge
+     *            any intersecting edge
+     * @param area
+     *            Area
+     * @return true if the edge is a valid intersecting edge
+     */
+    private boolean isValidIntersectingEdge(final Edge edge, final Area area)
+    {
+        return
+        // We do not want any pedestrian edges that are also pedestrian areas.
+        // Currently, pedestrian ways are ingested as both edges and as areas.
+        edge.getOsmIdentifier() != area.getOsmIdentifier() && edge.isMasterEdge()
+                && !HighwayTag.isPedestrianNavigableHighway(edge)
+                // Valid edge should be a car navigable highway
+                && (HighwayTag.isCarNavigableHighway(edge)
+                        // or a cycleway with no foot access
+                        || (Validators.isOfType(edge, HighwayTag.class, HighwayTag.CYCLEWAY)
+                                && Validators.isOfType(edge, FootTag.class, FootTag.NO))
+                        // or a service road that is not a yard, siding, spur or crossover
```

Comment:
nit: this comment doesn't seem to match the line below it?

Reply:
Thanks for pointing this out! Missed to update the comment after code update.